packages/now-go/util/analyze.go

```diff
@@ -209,6 +209,19 @@ func main() {
 					fmt.Print(string(analyzedJSON))
 					os.Exit(0)
 				}
+
+				// fallback, when ast coudn't parse, with multi-line comments
+				paramStr := fmt.Sprintf("%s", param.Type)
+				if strings.Contains(string(paramStr), "http ResponseWriter") {
```

Comment:
The above is more precise in term of checking type of params.

And I tried to check both, but `*http.Request` just gave me back pointer value instead `0xc0000cc4a0`, that why i'm go for this.

I think, we should rely on the the first condition. this is good only for fallback, when the above condition not work (edge case).

Reply:
@sophearak I believe the following source file will not work as expected.

Can you add a test for this behavior?

```go
package handler

import (
	"fmt"
	"net/http"
)

func Index1() {
	fmt.Fprintf(w, "one:RANDOMNESS_PLACEHOLDER")
}

func Index2(w http.ResponseWriter) {
	fmt.Fprintf(w, "two:RANDOMNESS_PLACEHOLDER")
}

func Index3(w http.ResponseWriter, req *http.Request) {
	fmt.Fprintf(w, "three:RANDOMNESS_PLACEHOLDER")
}
```

I would expect the output to be `three:RANDOMNESS_PLACEHOLDER`.

My guess is you will need to iterate over the functions twice. The first pass you should check for the exact match and the second pass can check for the fallback.

Reply:
@styfle Nice catch, I added more test for this.

However I `func Index1()`, since it will cause the build fail, due to syntax error. 
```
func Index1() {
	fmt.Fprintf(w, "one:RANDOMNESS_PLACEHOLDER")
}
```